### ISSUE TITLE
Add a Code of Conduct for the GeoPandas project

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#978]
I propose to adopt a Code of Conduct for the GeoPandas project, and more specifically based on the Contributor Covenant.

---

I *think* we never had a situation up to now where we would have needed to act upon one, but, you never need to wait until it is too late and it also the signal you give with it to community members and newcomers.
(it is also mentioned in https://github.com/geopandas/geopandas/community as recommended community standard, and in addition, it is also required to apply for NumFOCUS affiliated project)

I propose to add the Code of Conduct developed by the Contributor Covenant project (https://www.contributor-covenant.org/). It is also used by pandas, xarray, dask, ... 
But there are also some alternatives, eg the one of [Python](https://www.python.org/psf/codeofconduct/), [Jupyter](https://github.com/jupyter/governance/blob/master/conduct/code_of_conduct.md), ..

The main question is how to handle reports. The template expects you to fill in an email address. For pandas, we created a specific private google groups (that a "subcommittee" is subscribed to) for this. And I think in Dask they are doing the same, in addition of providing a form to report something anonymously (https://github.com/dask/governance/pull/10). 
We can do the same for GeoPandas.

cc @geopandas/collaborators @martinfleis @ljwolf

**Comment by story645** [comment]
Outsider here, but I like that the Jupyter one is the most concrete and has a whole enforcement protocol. Anonymous form + email sounds good (and let people know who's on the committee in case they have to report somebody on it).

**Comment by martinfleis** [comment]
I like the idea of having a Code of Conduct. Looking at examples above, I like the text of Contributor Covenant, but also like the Jupyter's explicit points. It seems to be much clearer than free flowing text one tends to skip. I slightly tend to prefer Jupyter, but I do not really mind, both are good.

Google group sounds like a good solution. I would be for including anonymous form as well, sometimes one might prefer anonymous report.

**Comment by jGaboardi** [comment]
FWIW, here is the [Code of Conduct](https://github.com/pysal/governance/blob/master/conduct/code_of_conduct.rst) we recently adopted in PySAL.

**Comment by jorisvandenbossche** [comment]
Coming back to this with much delay. First, thanks all for the input! 

Going through the different options, I actually also quite like the jupyter one. So maybe we can go with that one?

**Comment by jorisvandenbossche** [comment]
NumFOCUS also has its version of a code of conduct, which is somewhat similar to numpy's one (or the other way around, or they have a common source, they all seem to borrow from each other): https://numfocus.org/code-of-conduct 
So that's another option. Although I think I personally more like the list of behaviours of the jupyter one. I will try to do a PR with some proposal.

**Comment by jdmcbr** [comment]
@jorisvandenbossche I feel good about the jupyter CoC.

**Comment by nicole-numfocus** [comment]
@jorisvandenbossche -Ping to get this conversation moving again. NumFOCUS is ready to onboard Geopandas as soon as the CoC is added. We're happy to answer questions or provide context.

**Comment by martinfleis** [comment]
@jorisvandenbossche discussion here looks like we agreed on Jupyter version of CoC, so I'd go with that one. I can make a PR to adapt and add it over the weekend if there are no objections.

**Comment by jorisvandenbossche** [comment]
@nicole-numfocus Thanks for the gentle ping!

@martinfleis that's certainly welcome! Otherwise I can also look at it tomorrow. 
I started a branch some time ago (https://github.com/geopandas/geopandas/compare/master...jorisvandenbossche:code-of-conduct?expand=1), but that's just jupyter's version without any edits for geopandas yet (I started with that in order to easily see the changes).

From quickly looking at it again, some things that need to happen (apart from the obvious name changes):

- The intro probably needs to be edited / shortened somewhat (the first paragraph is a bit jupyter-specific)
- How to report: we can probably also create a google form and email address for this (similarly how jupyter does, and eg also dask: https://github.com/dask/governance/blob/master/code-of-conduct.md#reporting--enforcement)
  For this, we will also need a list of people who receive this
- Jupyter's version mentions an enforcement manual. I personally don't think we should maintain our own version of that, as other people are better positioned for that, so would rather refer to another source (eg just the one of Jupyter, or NumFOCUS also has one: https://numfocus.org/code-of-conduct/response-and-enforcement-events-meetups, although this might be more specific for events)
